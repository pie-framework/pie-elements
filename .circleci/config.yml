version: 2

defaults: &defaults
  working_directory: ~/repo
  docker:
    - image: circleci/node:11.10.0

reg_auth: &reg_auth
  name: Authenticate with registry
  command: echo "//registry.npmjs.org/:_authToken=$NPM_TOKEN" > ~/repo/.npmrc

restore_cache: &restore_cache
  keys:
    - v2-dependencies-{{ checksum "package.json" }}
    # fallback to using the latest cache if no exact match is found
    - v2-dependencies-

save_cache: &save_cache
  paths:
    - node_modules
  key: v2-dependencies-{{ checksum "package.json" }}

jobs:
  lint:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build lint
  test:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build babel test
  build:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build release --next --token $NPM_TOKEN
  release:
    <<: *defaults
    steps:
      - checkout
      - run: *reg_auth
      - restore_cache: *restore_cache
      - run: yarn install
      - save_cache: *save_cache
      - run: scripts/build release --token $NPM_TOKEN

workflows:
  version: 2

  simple_build:
    jobs:
      - lint:
          filters:
            branches:
              ignore:
                - develop
                - master
      - test:
          filters:
            branches:
              ignore:
                - develop
                - master
  build:
    jobs:
      - build:
          context: jira
          filters:
            branches:
              only:
                - develop

  release:
    jobs:
      - release:
          context: jira
          filters:
            branches:
              only:
                - master
  # nightly:
  #   triggers:
  #     - schedule:
  #         cron: "0 0 * * *"
  #         filters:
  #           branches:
  #             only:
  #               - develop
  #   jobs:
  #     - build
