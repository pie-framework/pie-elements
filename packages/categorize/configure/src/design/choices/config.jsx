import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import classNames from 'classnames';

import TextField from '@material-ui/core/TextField';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import MenuItem from '@material-ui/core/MenuItem';
import Menu from '@material-ui/core/Menu';
import { Checkbox } from '@pie-lib/config-ui';

const positionOptions = [
  {
    label: 'Above',
    value: 'above',
    secondaryText: 'Choices will be shown above categories'
  },
  {
    label: 'Below',
    value: 'below',
    secondaryText: 'Choices will be shown below categories'
  },
  {
    label: 'Left',
    value: 'left',
    secondaryText: 'Choices will be shown to the left of the categories'
  },
  {
    label: 'Right',
    value: 'right',
    secondaryText: 'Choices will be shown to the right of the categories'
  }
];

export class Config extends React.Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    className: PropTypes.string,
    config: PropTypes.object,
    categoryCountIsOne: PropTypes.bool,
    allChoicesHaveCount: PropTypes.func,
    onModelChanged: PropTypes.func
  };

  static defaultProps = {};

  state = {
    anchorEl: null
  };

  toggleRemoveAllTiles = () => {
    const { config, onModelChanged } = this.props;

    const allAtOne = this.props.allChoicesHaveCount(1);
    const update = config.choices.map(c => {
      c.categoryCount = allAtOne ? 0 : 1;
      return c;
    });

    onModelChanged({ choices: update });
  };

  changeLabel = ({ target }) => {
    this.props.onModelChanged({ choicesLabel: target.value });
  };

  toggleShuffle = () => {
    this.props.onModelChanged({
      lockChoiceOrder: !this.props.config.lockChoiceOrder
    });
  };

  changePosition = position => {
    this.props.onModelChanged({ choicesPosition: position.value });

    this.setState({ anchorEl: null });
  };

  handleClickPosition = event => {
    this.setState({ anchorEl: event.currentTarget });
  };

  handleMenuItemClick = (event, index) => {
    this.setState({ selectedIndex: index, anchorEl: null });
  };

  handleClose = () => {
    this.setState({ anchorEl: null });
  };

  render() {
    const { anchorEl } = this.state;
    const { classes, className, categoryCountIsOne, config } = this.props;

    const positionOption =
      positionOptions.find(option => option.value === config.choicesPosition) ||
      positionOption[1];

    return (
      <div className={classNames(classes.config, className)}>
        <div className={classes.row}>
          <Checkbox
            label={'Remove all tiles after placing'}
            checked={categoryCountIsOne}
            onChange={this.toggleRemoveAllTiles}
          />
          <Checkbox
            label={'Lock Choice Order'}
            checked={config.lockChoiceOrder}
            onChange={this.toggleShuffle}
          />
        </div>

        <div className={classes.configuration}>
          <TextField
            className={classes.label}
            InputLabelProps={{
              shrink: true
            }}
            label="Label"
            value={config.choicesLabel}
            onChange={this.changeLabel}
          />
          <List component="nav">
            <ListItem
              button
              aria-haspopup="true"
              onClick={this.handleClickPosition}
            >
              <ListItemText
                primary="Choices Position"
                secondary={positionOption.secondaryText}
              />
            </ListItem>
          </List>
          <Menu
            id="lock-menu"
            anchorEl={anchorEl}
            open={Boolean(anchorEl)}
            onClose={this.handleClose}
          >
            {positionOptions.map(option => (
              <MenuItem
                key={option.value}
                selected={option.value === config.choicesPosition}
                onClick={() => this.changePosition(option)}
              >
                {option.label}
              </MenuItem>
            ))}
          </Menu>
        </div>
      </div>
    );
  }
}
const styles = theme => ({
  configuration: {
    display: 'grid',
    alignItems: 'center',
    gridColumnGap: `${theme.spacing.unit}px`,
    gridRowGap: `${theme.spacing.unit}px`,
    gridTemplateColumns: 'repeat(2, 1fr)',
    justifyContent: 'space-between',
    marginBottom: theme.spacing.unit
  },
  choiceConfig: {
    width: '100%',
    maxWidth: 360,
    backgroundColor: theme.palette.background.paper
  },
  row: {
    display: 'grid',
    gridColumnGap: `${theme.spacing.unit}px`,
    gridRowGap: `${theme.spacing.unit}px`,
    gridTemplateColumns: 'repeat(2, 1fr)',
    justifyContent: 'space-between',
    marginTop: theme.spacing.unit,
    marginBottom: theme.spacing.unit
  },
  shuffleCheckbox: {
    gridColumnStart: 2,
    gridColumnEnd: 2
  }
});
export default withStyles(styles)(Config);
