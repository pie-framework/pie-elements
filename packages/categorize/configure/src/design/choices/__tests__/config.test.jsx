import { shallow } from 'enzyme';
import React from 'react';

import { Config } from '../config';


describe('config', () => {
  let onModelChanged;
  let allChoicesHaveCount;
  let config;

  beforeEach(() => {
    onModelChanged = jest.fn();
    allChoicesHaveCount = jest.fn();
    config = {
      choices: [
        {
          id: '0',
          content: 'Choice 0'
        },
      ],
      choicesPosition: 'below',
      choicesLabel: '',
      lockChoiceOrder: true,
      removeTilesAfterPlacing: false,
      categoriesPerRow: 2,
      categories: [
        {
          id: '0',
          label: 'Category 0',
          choices: []
        },
      ],
      rowLabels: [''],
      correctResponse: [],
      partialScoring: true,
    };
  });
  const wrapper = extras => {
    const props = { classes: {}, onModelChanged, allChoicesHaveCount, config, ...extras };
    return shallow(<Config {...props} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      const w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    describe('toggleRemoveAllTiles', () => {
      let w;

      it('adds categoryCount as 1 to choices if it was undefined', () => {
        w = wrapper();
        w.instance().toggleRemoveAllTiles();
        expect(onModelChanged).toBeCalledWith({ choices: [
          { id: '0', content: 'Choice 0', categoryCount: 1 }]});
      });

      it('sets categoryCount to 1 in every choice if it was 0', () => {
        w = wrapper({
          choices: [{ id: '0', content: 'Choice 0', categoryCount: 0 }]
        });
        w.instance().toggleRemoveAllTiles();
        expect(onModelChanged).toBeCalledWith({ choices: [
            { id: '0', content: 'Choice 0', categoryCount: 1 }]});
      });

      it('sets categoryCount to 0 in every choice if it was 1', () => {
        w = wrapper({
          choices: [{ id: '0', content: 'Choice 0', categoryCount: 1 }]
        });
        w.setProps({ allChoicesHaveCount: () => true });
        w.instance().toggleRemoveAllTiles();
        expect(onModelChanged).toBeCalledWith({ choices: [
            { id: '0', content: 'Choice 0', categoryCount: 0 }
          ]});
      });
    });

    it('changeLabel', () => {
      let w = wrapper();

      w.instance().changeLabel({ target: { value: 'foo' } });

      expect(onModelChanged).toBeCalledWith({
        choicesLabel: 'foo'
      });
    });

    it('toggleShuffle', () => {
      let w = wrapper();

      w.instance().toggleShuffle();

      expect(onModelChanged).toBeCalledWith({
        lockChoiceOrder: false
      });
    });

    it('changePosition', () => {
      let w = wrapper();

      w.instance().changePosition({ value: 'below' });

      expect(onModelChanged).toBeCalledWith({
        choicesPosition: 'below'
      });
    });

  });
});
