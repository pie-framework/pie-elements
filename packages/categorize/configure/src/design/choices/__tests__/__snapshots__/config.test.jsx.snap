// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config snapshot renders 1`] = `
<div
  className=""
>
  <div>
    <Component
      label="Remove all tiles after placing"
      onChange={[Function]}
    />
    <Component
      checked={true}
      label="Lock Choice Order"
      onChange={[Function]}
    />
  </div>
  <div>
    <TextField
      InputLabelProps={
        Object {
          "shrink": true,
        }
      }
      label="Label"
      onChange={[Function]}
      required={false}
      select={false}
      value=""
      variant="standard"
    />
    <WithStyles(List)
      component="nav"
    >
      <WithStyles(ListItem)
        aria-haspopup="true"
        button={true}
        onClick={[Function]}
      >
        <WithStyles(ListItemText)
          primary="Choices Position"
          secondary="Choices will be shown below categories"
        />
      </WithStyles(ListItem)>
    </WithStyles(List)>
    <WithStyles(Menu)
      anchorEl={null}
      id="lock-menu"
      onClose={[Function]}
      open={false}
    >
      <WithStyles(MenuItem)
        key="above"
        onClick={[Function]}
        selected={false}
      >
        Above
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="below"
        onClick={[Function]}
        selected={true}
      >
        Below
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="left"
        onClick={[Function]}
        selected={false}
      >
        Left
      </WithStyles(MenuItem)>
      <WithStyles(MenuItem)
        key="right"
        onClick={[Function]}
        selected={false}
      >
        Right
      </WithStyles(MenuItem)>
    </WithStyles(Menu)>
  </div>
</div>
`;
